### ISSUE TITLE
Syslog depends on unmaintained library

### ISSUE THREAD

**Issue by SamWhited** [moby/moby#39962]
The syslog functionality currently depends on https://github.com/RackSec/srslog which is unmaintained. Because the library is vendored this doesn't present an immediate problem, but if any major issue is ever discovered with it in the future we don't want to be scrambling to upgrade to an entirely new library.

It may be worth looking into a different syslog library such as https://github.com/hashicorp/go-syslog (just a simple wrapper around `log/syslog`, does not support Windows which may be a requirement), or some other library.

**Comment by cpuguy83** [comment]
The reason we use srslog is because `log/syslog` is also unmaintained :(

**Comment by thaJeztah** [comment]
Looks like @RackSec archived _all_ their repositories; did they perhaps move them elsewhere? Or just stopped doing open source?

**Comment by SamWhited** [comment]
> The reason we use srslog is because log/syslog is also unmaintained :(

`log/syslog` is in the standard library so it will at least get security fixes and the like if any major problems were discovered, if I'm not mistaken.

> Looks like @RackSec archived all their repositories; did they perhaps move them elsewhere? Or just stopped doing open source?


I didn't see another version of this while searching on GoDoc, and there are no comments on their profile or any of their repos that I looked at. Maybe @irinarenteria or @szilveszter can chime in and clarify? Thanks!

**Comment by jas4711** [comment]
Hi!  I help maintain Go packages in Debian.  I found that moby uses github.com/RackSec/srslog which is archived.  Please use some other package instead, we don't want archived projects as dependencies.

**Comment by thaJeztah** [comment]
Yeah, have looked at that dependency a few times, and considered that as a (temporary) step we should fork it as an internal package. That keeps the status quo while we look if there's better replacements, and at least allows us to fix issues in the code if there's any.

**Comment by thaJeztah** [comment]
- opened https://github.com/moby/moby/pull/53477 to move it to an internal package.
